lms/djangoapps/instructor/views/api.py

```diff
@@ -3560,3 +3557,91 @@ def _get_certificate_for_user(course_key, student):
+    """
+    Checks and retrieves a customized "from address", if one exists for the course/org. This is the email address that
+    learners will see the message coming from.
+
+    Args:
+        course_overview (CourseOverview): The course overview instance for the course-run.
+
+    Returns:
+        String: The customized "from address" to be used in messages sent by the bulk course email tool for this
+                course or org.
+    """
+    from_addr = configuration_helpers.get_value('course_email_from_addr')
+    if isinstance(from_addr, dict):
+        # If course_email_from_addr is a dict, we are customizing the email template for each organization that has
+        # courses on the site. The dict maps from addresses by org allowing us to find the correct from address to use
+        # here.
+        from_addr = from_addr.get(course_overview.display_org_with_default)
+
+    return from_addr
+
+
+def _get_branded_email_template(course_overview):
+    """
+    Checks and retrieves the custom email template, if one exists for the course/org, to style messages sent by the bulk
+    course email tool.
+
+    Args:
+        course_overview (CourseOverview): The course overview instance for the course-run.
+
+    Returns:
+        String: The name of the custom email template to use for this course or org.
+    """
+    template_name = configuration_helpers.get_value('course_email_template_name')
+    if isinstance(template_name, dict):
+        # If course_email_template_name is a dict, we are customizing the email template for each organization that has
+        # courses on the site. The dict maps template names by org allowing us to find the correct template to use here.
+        template_name = template_name.get(course_overview.display_org_with_default)
+
+    return template_name
+
+
+def _convert_schedule_to_utc_from_local(schedule, browser_timezone, user):
+    """
+    Utility function to help convert the schedule of an instructor task from the requesters local time and timezone
+    (taken from the request) to a UTC datetime.
+
+    Args:
+        schedule (String): The desired time to execute a scheduled task, in local time, in the form of an ISOString.
+        timezone (String): The time zone, as captured by the user's web browser, in the form of a string.
+        user (User): The user requesting the action, captured from the originating web request. Used to lookup the
+                     the time zone preference as set in the user's account settings.
+
+    Returns:
+        DateTime: A datetime instance describing when to execute this schedule task converted to the UTC timezone.
+    """
+    # look up the requesting user's timezone from their account settings
+    preferred_timezone = get_user_preference(user, 'time_zone', username=user.username)
+    # use the user's preferred timezone (if available), otherwise use the browser timezone.
+    timezone = preferred_timezone if preferred_timezone else browser_timezone
```

Comment:
I was thinking that we would send the browser timezone with the scheduled requests as the fallback option in case there is no preferred timezone set. I'm not too familiar with potential issues with grabbing the timezone from a browser.

I did try to protect against a situation where we don't have either timezone. We will raise a `ValueError` if we don't have a timezone and stop trying to schedule the request.

Reply:
sounds good!